packages/owl-bot/test/owl-bot.ts

```diff
@@ -1738,6 +1745,213 @@ describe('owlBot', () => {
   });
 });
 
+describe('locking behavior', () => {
+  let probot: Probot;
+  beforeEach(async () => {
+    sandbox.stub(process, 'env').value({
+      APP_ID: '1234354',
+      PROJECT_ID: 'foo-project',
+      CLOUD_BUILD_TRIGGER: 'aef1e540-d401-4b85-8127-b72b5993c20d',
+      CLOUD_BUILD_TRIGGER_REGENERATE_PULL_REQUEST:
+        'aef1e540-d401-4b85-8127-b72b5993c20e',
+    });
+    probot = createProbot({
+      overrides: {
+        githubToken: 'abc123',
+        Octokit: ProbotOctokit.defaults({
+          retry: {enabled: false},
+          throttle: {enabled: false},
+        }),
+      },
+    });
+    await probot.load((app: Probot) => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      owlbot.OwlBot('abc123', app, sandbox.stub() as any);
+    });
+  });
+  afterEach(() => {
+    sandbox.restore();
+  });
+  it('returns immediately if lock cannot be acquired', async () => {
```

Comment:
Maybe we should assert a test where we do create the lock, and we release?

Reply:
I added tests ensuring `release` is called (which I think is valuable). I didn't bother testing the behavior of an event being triggered after we call `.release`, as I don't think this gives us any additional coverage (_since the acquireLock method is mocked_).